src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1635,21 +1649,16 @@ namespace winrt::Microsoft::Terminal::Control::implementation
 
     void ControlCore::BlinkAttributeTick()
     {
-        auto lock = _terminal->LockForWriting();
+        const auto lock = _terminal->LockForWriting();
 
         auto& renderSettings = _terminal->GetRenderSettings();
         renderSettings.ToggleBlinkRendition(*_renderer);
     }
 
     void ControlCore::BlinkCursor()
     {
-        if (!_terminal->IsCursorBlinkingAllowed() &&
-            _terminal->IsCursorVisible())

```

Comment:
BTW this if condition is incorrect. If you flip the early return this old code says:
```cpp
if (!(!_terminal->IsCursorBlinkingAllowed() && _terminal->IsCursorVisible()))
    _terminal->SetCursorOn(!_terminal->IsCursorOn());
```
which is:
```cpp
if (_terminal->IsCursorBlinkingAllowed() || !_terminal->IsCursorVisible())
    _terminal->SetCursorOn(!_terminal->IsCursorOn());
```

In other words, this old code would blink the cursor even if it's invisible.